packages/components/src/box-control/index.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import { useInstanceId } from '@wordpress/compose';
+import { useState, useRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
+import { Flex } from '../flex';
+import AllInputControl from './all-input-control';
+import InputControls from './input-controls';
+import BoxControlIcon from './icon';
+import Text from '../text';
+import LinkedButton from './linked-button';
+import Visualizer from './visualizer';
+import {
+	Root,
+	Header,
+	HeaderControlWrapper,
+} from './styles/box-control-styles';
+import { useBoxControlState, isValuesMixed } from './utils';
+
+const defaultInputProps = {
+	min: 0,
+};
+
+function useUniqueId( idProp ) {
+	const instanceId = useInstanceId( BoxControl );
+	const id = `inspector-box-control-${ instanceId }`;
+
+	return idProp || id;
+}
+export default function BoxControl( {
+	id: idProp,
+	inputProps = defaultInputProps,
+	onChange = noop,
+	label = __( 'Box Control' ),
+	values: valuesProp,
+	units,
+} ) {
+	const [ values, setValues ] = useBoxControlState( valuesProp );
```

Comment:
I'm wondering if `useControlledState` isn't too much abstraction, causing some indirection here.

Also, `useBoxControlState` is really just `useControlledState` with a particular default. I would avoid it until there's a specific need to reuse that pattern.